libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java

```diff
@@ -489,7 +488,9 @@ private static ScheduledFuture createPublishHandler() {
 
             @Override
             public void run() {
-                AnalyticsPublisherService.startActionPublish(SalesforceSDKManager.getInstance().getAppContext());
+                AnalyticsPublishingWorker.Companion.startActionPublish(
+                        SalesforceSDKManager.getInstance().getAppContext()
+                );
             }
         };
         return scheduler.scheduleAtFixedRate(publishRunnable, 0, sPublishFrequencyInHours, TimeUnit.HOURS);
```

Comment:
@brandonpage - While I was really thinking this change through, doesn't it seem like this `ScheduledExecutorService` is potentially redundant if it's now running a background tasks `OneTimeWorkRequest`?  Since this wants to publish at periodic intervals, what if we replaced this with `PeriodicWorkRequest` at this level? Maybe I ought to take this change a little further.

Even just typing it out loud, I'm feeling that's what we want to do so we're not laying new scheduling code under the old one here 💡 I'll give it a whirl!

Reply:
That looks a lot better, even in the Java code since it doesn't have to worry about cancellation anymore.  See 👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2401/commits/2ea90b39ec9aed414485805b7b67e0a72ddf94e9